### PR TITLE
Deployment component feature addition to handle base model deployment

### PR DESCRIPTION
**Pull request by HrishikeshGeedMS**
This PR contains, updates for deployment component to handle the direct deployment for base models .

### CODE REVIEW

File: common/src/model_deployment/deploy.py [hunk 1]
```diff
@@ -296,7 +319,7 @@ def main(args):
         endpoint_name=endpoint_name,
         deployment_name=deployment_name,
         model_id=model_id,
-        args=args
+        args=args,
```
Comment (skasturi) [inline common/src/model_deployment/deploy.py:322]: Please log all the args.

**Comment by skasturi** [comment]
Are three more changes in addition to #805 ?

File: common/src/model_deployment/deploy.py (hunk 1 above)
Comment (HrishikeshGeedMS) [reply common/src/model_deployment/deploy.py]: Yes, generating those in other seperate PR which includes logging details for all the common component
i.e registration, deployment and local_validation

File: assets/common/components/deploy_model/spec.yaml [hunk 2]
```diff
@@ -39,9 +40,14 @@ inputs:
   # Output of registering component
   registration_details:
     type: uri_file
-    optional: false
+    optional: true
     description: Json file that contains the ID of registered model to be deployed
 
+  model_id:
```
Comment (novaturient95) [inline assets/common/components/deploy_model/spec.yaml:46]: Please add more details in the description

**Review by novaturient95 — APPROVED** [review]

**Review by novaturient95 — DISMISSED** [review]

File: assets/common/components/deploy_model/spec.yaml [hunk 3]
```diff
@@ -39,9 +40,17 @@ inputs:
...
+    type: string
+    optional: true 
+    description: |
+      Asset ID of the model registered in workspace/registry. Asset ID format looks like below.\
```
Comment (novaturient95) [inline assets/common/components/deploy_model/spec.yaml:50]: `\` at end is not needed here. Just make sure to give a space at the end of previous line.

Comment (HrishikeshGeedMS) [reply assets/common/components/deploy_model/spec.yaml]: Done

**Review by novaturient95 — APPROVED** [review]

File: assets/common/components/deploy_model/spec.yaml (hunk 3 above)
Comment (vizhur) [inline assets/common/components/deploy_model/spec.yaml]: AssetId format. "remove looks like below"

File: assets/common/src/model_deployment/deploy.py [hunk 4]
```diff
@@ -255,11 +261,17 @@ def main(args):
...
+        model_info = {}
+        with open(args.registration_details) as f:
+            model_info = json.load(f)
+        model_id = model_info["id"]
+        model_name = model_info["name"]
+    elif args.model_id:
+        model_id = str(args.model_id)
+        model_name = model_id.split("/")[-3]
+    else:
+        raise Exception("Arguments model_id and registration_details both are missing.")
```
Comment (skasturi) [inline assets/common/src/model_deployment/deploy.py:274]: Are we integrating logs in a different PR?

Comment (HrishikeshGeedMS) [reply assets/common/src/model_deployment/deploy.py:274]: Yes sasi. PR containing logs has lots of changes , so that is in different pr

**Review by vizhur — APPROVED** [review]

**Review by novaturient95 — APPROVED** [review]

**Pull request merged**